src/states/away/CountdownToAwayNotReady.java

```diff
@@ -1,7 +1,63 @@
 package states.away;
 
+import events.CheckAllZones;
+import events.TimerRanOut;
 import states.Countdown;
+import states.NotReady;
+import states.SecurityContext;
+import timer.Timer;
 
 public class CountdownToAwayNotReady extends Countdown {
 
+	private static CountdownToAwayNotReady instance;
+
+	/**
+	 * Private for the singleton pattern
+	 */
+	private CountdownToAwayNotReady() {
+	}
+
+	/**
+	 * For singleton
+	 * 
+	 * @return the object
+	 */
+	public static CountdownToAwayNotReady instance() {
+		if (instance == null) {
+			instance = new CountdownToAwayNotReady();
+		}
+		return instance;
+	}
+
+	/**
+	 * Processes a timer tick, generates a Timer Ran Out event
+	 */
+	@Override
+	public void handleEvent(TimerRanOut event) {
+		SecurityContext.instance().showSecondsToAway(0);
+		SecurityContext.instance().changeState(NotReady.instance());
+	}
+
+	/**
+	 * Processes all the zones being checked
+	 */
+	@Override
+	public void handleEvent(CheckAllZones event) {
+
+		SecurityContext.instance().changeState(CountdownToAway.instance());
+	}
+
+	@Override
+	public void enter() {
+		timer = new Timer(this, 10);
```

Comment:
Have you tried and run this code? I have a feeling, that between your states CountdownToAway and CountdownToAwayNotReady, whenever you CheckAllZones or UncheckZone, a new 10-second timer always starts... Try and run it on your computer.

Reply:
Yes, this problem is fixed in another branch. It is ready for PR whenever this one gets approved